### DISCUSSION TITLE
4M Sats - LNURL-Withdraw support (LN Wallet read LNURL-withdraw)

### DISCUSSION THREAD

**Discussion by pretyflaco** [GaloyMoney/galoy#1417] [Bounties]
## Description
Implement LNURL-withdraw support which will allow user `LN Wallet` to read LNURL-withdraw QR or string to receive funds from a `LN Service`. 

## Background

Typically when a user wishes to receive sats, they generate an invoice, and that invoice is paid by the sender. In most user-facing b2c applications, it is pretty bad UX for the user to have to generate an invoice and then paste it into a service. This process is particularly tricky when users try to withdraw funds into mobile wallets while using a different device e.g. desktop website or Lightning ATM and have to communicate the generated invoice to `LN Service`.

The LNURL-withdraw flow standardises the communication of this LN invoice and payment of the invoice into a single UX action initiated by the `LN Wallet`. 

Useful docs:
[LUD-03](https://github.com/fiatjaf/lnurl-rfc/blob/luds/03.md)
[LUD-08](https://github.com/fiatjaf/lnurl-rfc/blob/luds/08.md)
[LUD-17](https://github.com/fiatjaf/lnurl-rfc/blob/luds/17.md)
[LNPay Docs](https://docs.lnpay.co/api/lnurl-withdraw/how-does-lnurl-withdraw-work)
[BOLT.FUN Docs](https://bolt.fun/guide/web-services/lnurl/withdraw)

## Task
Submit a PR that will allow users to receive funds to their account from `LN Service` by scanning a LNURL-withdraw QR or pasting LNURL-withdraw string 

## Definition of Done
PR gets merged and enables user to withdraw funds from `LN Service` by scanning LNURL-withdraw QR or pasting LNURL-withdraw string.

## Bounty
4M Sats (0.04 BTC)

## Bounty Patrons
[Galoy](https://galoy.io/)

**Comment by kngako** [comment]
Anyone working on this?

**Reply by pretyflaco** [reply]
Thank you for the PR @kngako - it's currently in testing. Will update shortly.
